intellij/src/saros/intellij/runtime/FilesystemRunner.java

```diff
@@ -75,16 +53,13 @@ private FilesystemRunner() {
    * @see Application#runWriteAction(Runnable)
    */
   public static void runWriteAction(
-      @NotNull final Runnable runnable, @Nullable final ModalityState modalityState) {
-
-    final ModalityState chosenModalityState =
```

Comment:
Why is this check not necessary anymore?

Reply:
Because the new logic does not accept null values. The field is now marked as `NotNull`.

The old implementation was just a different approach to offer a convenience call: Specify the modality state or, if you don't (i.e. pass `null`), we will use the default modality state. I thought it would be clearer if there rather were two different methods, one requiring a modality state (that is non-null) and one that does not (and always uses the default modality state).

I also checked all calls of these methods to ensure that none of them are passing null when calling one of these methods.

Also, as no call ever passed null, I did not introduce any such convenience methods here. But that goes back to the other point of whether it is necessary to replace all old calls that use the default modality state with the convenience method.